feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,625 @@
+	portId       = *ygot.Uint32(10)
+	deviceId     = *ygot.Uint64(1)
+	inId0        = *ygot.Uint64(0)
+	inId90       = *ygot.Uint64(90)
+	inId100      = *ygot.Uint64(100)
+	inId101      = *ygot.Uint64(101)
+	inId102      = *ygot.Uint64(102)
+	streamName   = "p4rt"
+)
+
+// Define PacketIO Interface
+type PacketIO interface {
+	GetTableEntry(delete bool) []*p4rtutils.ACLWbbIngressTableEntryInfo
+}
+
+// Define GDP Packet io structure
+type GDPPacketIO struct {
+	PacketIOPacket
+	IngressPort string
+}
+
+// Define PacketIO Structure
+type PacketIOPacket struct {
+	EthernetType *uint32
+}
+
+// Define Multiple Client Args Structure
+type mcTestArgs struct {
+	name     string
+	Items    [2]*testArgs
+	wantFail bool
+}
+
+// Define Test Args Structure
+type testArgs struct {
+	desc            string
+	lowID           uint64
+	highID          uint64
+	deviceID        uint64
+	handle          *p4rt_client.P4RTClient
+	wantStatus      int32
+	wantFinalStatus int32
+	wantFail        bool
+	wantWrite       bool
+	wantRead        bool
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
+	component := &oc.Component{}
+	component.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	component.Name = ygot.String(*p4rtNodeName)
+	component.IntegratedCircuit.NodeId = ygot.Uint64(deviceId)
+	gnmi.Replace(t, dut, gnmi.OC().Component(*p4rtNodeName).Config(), component)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	portName := dut.Port(t, "port1").Name()
+	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+}
+
+// Create client connection
+func clientConnection(t *testing.T, dut *ondatra.DUTDevice) *p4rt_client.P4RTClient {
+	clientHandle := p4rt_client.P4RTClient{}
+	if err := clientHandle.P4rtClientSet(dut.RawAPIs().P4RT().Default(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+	return &clientHandle
+}
+
+// Setup P4RT Arbitration Stream which will return a GRPC status code
+// in line with the client arbitration notifications. An error can also
+// be returned.
+func streamP4RTArb(args *testArgs) (int32, error) {
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    args.deviceID,
+		ElectionIdH: args.highID,
+		ElectionIdL: args.lowID,
+	}
+	// Send arb msg
+	if args.handle != nil {
+		args.handle.StreamChannelCreate(&streamParameter)
+		if err := args.handle.StreamChannelSendMsg(&streamName, &p4_v1.StreamMessageRequest{
+			Update: &p4_v1.StreamMessageRequest_Arbitration{
+				Arbitration: &p4_v1.MasterArbitrationUpdate{
+					DeviceId: streamParameter.DeviceId,
+					ElectionId: &p4_v1.Uint128{
+						High: streamParameter.ElectionIdH,
+						Low:  streamParameter.ElectionIdL,
+					},
+				},
+			},
+		}); err != nil {
+			return 0, errors.New("Errors seen when sending ClientArbitration message.")
+		}
```

Comment:
You can now get the StreamTermErr after sendinding Msg.

```
if _, _, arbErr := client.StreamChannelGetArbitrationResp(&streamName, 1); arbErr != nil {
  if err := p4rtutils.StreamTermErr(client.StreamTermErr); err != nil {
	  return err
  }
  return fmt.Errorf("errors seen in ClientArbitration response: %v", arbErr)
}
```

Reply:
I tried to look at `StreamTermErr` but it seemed like it was `nil`. Is it possible that the stream gets destroyed before reaching the code? 
Here's a view from DLV in VSCode of the args
![image](https://user-images.githubusercontent.com/64392846/212803505-cc293099-bd84-4460-86d1-fb3ed9969f27.png)
 
Code Snippet (don't mind the `return 3`, I was putting it there for debug purposes.
```go
func getRespCode(args *testArgs) (int32, error) {
	handle := args.handle
	if handle != nil {
		// Grab Arb Response to look at status code
		_, arbResp, arbErr := handle.StreamChannelGetArbitrationResp(&streamName, 1)
		if arbErr != nil {
			if err := p4rtutils.StreamTermErr(args.handle.StreamTermErr); err != nil {
				return 3, err
			}
			return 0, fmt.Errorf("Errors seen in ClientArbitration response: %v", arbErr)
		}
		// Handle exception when GetCode is empty.
		respCode := arbResp.Arb.GetStatus().GetCode()
		return respCode, nil
	}
	return 0, errors.New("Missing Client")
}
```

Reply:
I think it is because of how you initialize `P4RTClient`.

* If you do `p4rt_client.P4RTClient{}` then `StreamTermErr` channel is not initialized.

The proper way would be using the method `p4rt_client.NewP4RTClient()`.